standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -1257,6 +1257,10 @@ public enum ConfVars {
 
 
     // Partition management task params
+    PARTITION_MANAGEMENT_ENABLE("metastore.partition.management.enable", "hive.metastore.partition.management.enable",
```

Comment:
how about setting the `metastore.partition.management.task.frequency` to 0 to disable the task?

Reply:
+1, but maybe -1 as default

Reply:
If we set it 0 or -1 then we didn't handle else condition properly here: https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/leader/HouseKeepingTasks.java#L116-L118
`long freq = task.runFrequency(TimeUnit.MILLISECONDS);
  runningTasks.add(task);
  metastoreTaskThreadPool.getPool().scheduleAtFixedRate(task, freq, freq, TimeUnit.MILLISECONDS);`
  and scheduleAtFixedRate() would throw illegalArgumentException.
  
  Also, I thought a separate config would make it clear for end-users. If you think there are too many configs and we should only work with `metastore.partition.management.task.frequency`, please let me know. Thanks.

Reply:
I would prefer the `metastore.partition.management.task.frequency`, simply making it as 0 or -1, can achieve the same purpose with a new property setting as false explicitly, without adding extra document to train the user.

We should add the `freq > 0` validation in `HouseKeepingTasks`.

Reply:
Ack